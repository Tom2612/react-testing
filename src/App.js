import React, { useEffect, useState } from 'react';
import User from './User';
import Input from './Input';

const App = () => {
    const [user, setUser] = useState(null);
    const [error, setError] = useState('');

    const [counter, setCounter] = useState(0);

    const [inputValue, setInputValue] = useState('');

    const handleChange = (event) => {
        setInputValue(event.target.value);
    };

    const increment = () => {
        setCounter(prev => ++prev)
    };

    const decrement = () => {
        setCounter(prev => --prev)
    };

    // useEffect(() => {
    //     fetch('https://jsonplaceholder.typicode.com/users/1')
    //   .then((response) => response.json())
    //   .then((user) => setUser(user))
    //   .catch((error) => setError(error.message));
    // }, []);

    if(error){
        return <span>{error}</span>
    }
    
    return (
        <div>
            <div>{user ? <User user={user} /> : <span>Loading...</span>}</div>
            <h2 data-testid="counter">{counter}</h2>
            <button onClick={decrement}>Decrement</button>
            <button onClick={increment}>Increment</button>
            <Input handleChange={handleChange} inputValue={inputValue} />
        </div>
    );  
};

export default App;