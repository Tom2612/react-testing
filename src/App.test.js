import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import userEvent from '@testing-library/user-event';

// describe("App component", () => {
//   it("renders correct heading", () => {
//     const { getByRole } = render(<App />);
//     expect(getByRole("heading").textContent).toMatch(/our first test/i);
//   });
// });

describe("App component", () => {
  it('renders Magnificent Monkeys', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders radical rhinos after button click", () => {
    render(<App />);
    const button = screen.getByRole("button", { name: 'Click me' });

    userEvent.click(button);

    expect(screen.getByRole("heading").textContent).toMatch(/radical rhinos/i);
  });
});