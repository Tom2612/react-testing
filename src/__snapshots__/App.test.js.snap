// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders Magnificent Monkeys 1`] = `
<div>
  <button
    type="button"
  >
    Click me
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
  ;
</div>
`;

exports[`App component renders magnificent monkeys 1`] = `
<div>
  <button
    type="button"
  >
    CLick me
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
  ;
</div>
`;
